frontend/packages/integration-tests-cypress/support/login.ts

```diff
@@ -1,59 +1,76 @@
-    cy.clearCookie('openshift-session-token');
+Cypress.Commands.add(
+  'login',
+  (
+    provider: string = 'kube:admin',
+    username: string = 'kubeadmin',
+    password: string = Cypress.env('BRIDGE_KUBEADMIN_PASSWORD'),
+  ) => {
+    cy.session(
+      [provider, username],
+      () => {
+        cy.visit('/'); // visits baseUrl which is set in plugins/index.js
+        cy.window().then((win: any) => {
+          // Check if auth is disabled (for a local development environment)
+          if (win.SERVER_FLAGS?.authDisabled) {
+            cy.task('log', '  skipping login, console is running with auth disabled');
+            return;
+          }
 
-    const idp = provider || KUBEADMIN_IDP;
-    cy.task('log', `  Logging in as ${username || KUBEADMIN_USERNAME}`);
-    cy.byLegacyTestID('login').should('be.visible');
-    cy.get('body').then(($body) => {
-      if ($body.text().includes(idp)) {
-        cy.contains(idp).should('be.visible').click();
-      }
-    });
-    cy.get('#inputUsername').type(username || KUBEADMIN_USERNAME);
-    cy.get('#inputPassword').type(password || Cypress.env('BRIDGE_KUBEADMIN_PASSWORD'));
-    cy.get(submitButton).click();
-    masthead.username.shouldBeVisible();
-  });
-});
-
-Cypress.Commands.add('logout', () => {
-  // Check if auth is disabled (for a local development environment).
-  cy.window().then((win: any) => {
-    if (win.SERVER_FLAGS?.authDisabled) {
-      cy.task('log', '  skipping logout, console is running with auth disabled');
-      return;
-    }
-    cy.task('log', '  Logging out');
-    cy.byTestID('user-dropdown').click();
-    cy.byTestID('log-out').should('be.visible');
-    cy.byTestID('log-out').click({ force: true });
-  });
-});
+          if (isLocalDevEnvironment) {
+            cy.exec(
+              'oc get route oauth-openshift -n openshift-authentication -o json | jq .spec.host',
+            ).then((result) => {
+              Cypress.env('OAUTH_BASE_ADDRESS', result.stdout.replace(/"/g, ''));
+              cy.origin(
+                Cypress.env('OAUTH_BASE_ADDRESS'),
+                { args: { provider, username, password } },
+                // eslint-disable-next-line @typescript-eslint/no-shadow
+                ({ provider, username, password }) => {
+                  // note required duplication in else below due to limitations of cy.origin
+                  cy.task('log', `  Logging in as ${username}`);
+                  cy.get('[data-test-id="login"]').should('be.visible');
+                  cy.get('body').then(($body) => {
+                    if ($body.text().includes(provider)) {
+                      cy.contains(provider).should('be.visible').click();
+                    }
+                  });
+                  cy.get('#inputUsername').type(username);
+                  cy.get('#inputPassword').type(password);
+                  cy.get('button[type=submit]').click();
+                },
+              );
```

Comment:
Could we pull this out into a view?

Reply:
No because of the limitation noted in the comment on line 43.  It's possible there is a way, but I couldn't figure out what it was...